src/App.tsx

```diff
@@ -15,36 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_ALPHABETICALLY = 'Sort alphabetically',
+  SORT_BY_LENGTH = 'Sort by length',
+}
+
+interface SortedParams {
+  sortField: string;
+  reversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reversed }: SortedParams,
+) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      if (sortField === SortType.SORT_ALPHABETICALLY) {
+        return good1.localeCompare(good2);
+      }
+
+      if (sortField === SortType.SORT_BY_LENGTH) {
+        return good1.length - good2.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (reversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
+
+  const resetSorting = () => {
+    setSortField('');
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          onClick={() => setSortField(SortType.SORT_ALPHABETICALLY)}
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.SORT_ALPHABETICALLY,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          onClick={() => setSortField(SortType.SORT_BY_LENGTH)}
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortType.SORT_BY_LENGTH,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          onClick={() => setReversed(!reversed)}
+          className={cn('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {sortField || reversed ? (
+          <button
+            type="button"
+            onClick={resetSorting}
+            className="button is-danger"
+          >
+            Reset
+          </button>
+        ) : null}
```

Comment:
```suggestion
        {hasResetButton && (
          <button
            type="button"
            onClick={resetSorting}
            className="button is-danger"
          >
            Reset
          </button>
        )}
```
- You don't need to use the ternary operator if you return nothing in the second case
- Create a variable for `sortField || reversed` to avoid mixing conditions